src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -866,6 +868,25 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
         }
     }
 
+    /**
+     * In the case we follow an elected master the new cluster state needs to have the same elected master and
+     * the new cluster state version needs to be equal or higher than our cluster state version. If either conditions
+     * are true then the cluster state is dated and we should ignore it.
+     */
+    private boolean newClusterStateDated(ClusterState currentState, ClusterState newClusterState) {
+        if (currentState.nodes().masterNodeId() != null) {
+            if (!currentState.nodes().masterNodeId().equals(newClusterState.nodes().masterNodeId())) {
+                logger.warn("received a cluster state that has [{}] master node, while current cluster state has [{}] as master node, ignoring...", newClusterState.nodes().masterNode(), currentState.nodes().masterNode());
```

Comment:
can we call this log:

```
received a cluster state from a different master then the current one, ignoring (received {}, current {}) 
```

also note that disco nodes already have [] in their toString.